crates/examples/infra/mod.rs

```diff
@@ -929,24 +963,26 @@ pub async fn main_entry_point<
                 <RandomBuilderImplementation as TestBuilderImplementation<TYPES>>::start(
                     run_config.config.num_nodes_with_stake.into(),
                     run_config.random_builder.clone().unwrap_or_default(),
+                    HashMap::new(),
                 )
                 .await;
 
-            run_config.config.builder_url = builder_url;
+            run_config.config.builder_urls = vec1::vec1![builder_url];
```

Comment:
For an example, it seems like vec1 could potentially be overkill? If during initialization we always pass a builder url, and presumably die if one is received, is it necessarily required to have hardline enforcement like this?

Reply:
We don't always re-assign the builder URL - for example, in case of External builder
My reasoning was:
- To run, HotShot needs at least one builder URL, hard requirement
- Vec1 represents this more accurately
- Vec1 allows for less manual checks for whether builder URLs are non-empty, especially in deserialization of HS config and in examples, where we do some complex manipulations with `run_config`

I'm not particularly attached to Vec1 here if you think there are reasons to avoid it.